manila/share/drivers/netapp/dataontap/cluster_mode/lib_base.py

```diff
@@ -3124,6 +3124,8 @@ def promote_replica(self, context, replica_list, replica, access_rules,
                 constants.STATUS_ERROR)
             new_active_replica['status'] = constants.STATUS_ERROR
             return [new_active_replica]
+        except exception.ReplicationException as e:
+            LOG.exception("Could not promote replica %s. %s", replica['id'], e)
```

Comment:
I think this should get the same treatment like the `exception.StorageCommunicationException` which is caught directly above, i.e. setting the replica to error and return

Reply:
any idea how to populate the error to user in driver, we have to use the message api?